### PR TITLE
Support for custom image URL

### PR DESCRIPTION
**Pull request by WanQiyang**
This adds support for custom URL prefix after uploading. It is useful when using CDN or in other situations where it is not possible to access the images directly through the default S3 URL.

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -63,6 +64,7 @@ const DEFAULT_SETTINGS: S3UploaderSettings = {
 	useCustomEndpoint: false,
 	customEndpoint: "",
 	forcePathStyle: false,
+	customImageUrl: "",
```
Comment (jvsteiner) [inline main.ts:67]: I prefer to use an explicit bool setting like with `customEndpoint` and `useCustomEndpoint` where we check whether the setting should be used, and only then invoke the required logic.  As it is in this PR, the use of the customUrl is implicit based on whether the value is "" or not.  This relies on the falsyness of "", which is not explicit, and inconsistent with the existing implementation.

File: main.ts [hunk 2]
```diff
@@ -292,7 +304,9 @@ export default class S3UploaderPlugin extends Plugin {
 		let apiEndpoint = this.settings.useCustomEndpoint
 			? this.settings.customEndpoint
 			: `https://s3.${this.settings.region}.amazonaws.com/`;
-		this.settings.imageUrlPath = this.settings.forcePathStyle
+		this.settings.imageUrlPath = this.settings.customImageUrl.length > 0
```
Comment (jvsteiner) [inline main.ts:307]: this is where the check of the setting should take place explicitly

**Comment by jvsteiner** [comment]
Thanks for this, I am happy to merge, once the comments have been addressed

**Comment by WanQiyang** [comment]
Thank you and I have added a new boolean option "useCustomImageUrl" in the latest commit to control this feature.

**Comment by jvsteiner** [comment]
Could you please make it so that these changes are all in a single commit, rather than in three separate ones?  you will probably need to put them on their own branch (on your fork) and raise a new request.

**Comment by jvsteiner** [comment]
I'd rather you didn't change the version number

**Pull request closed without merging**